src/transformers/trainer.py

```diff
@@ -1170,6 +1178,74 @@ def get_optimizer_cls_and_kwargs(args: TrainingArguments) -> Tuple[Any, Any]:
             optimizer_cls = torch.optim.Adagrad
         elif args.optim == OptimizerNames.RMSPROP:
             optimizer_cls = torch.optim.RMSprop
+        elif args.optim in [
+            OptimizerNames.GALORE_ADAMW,
+            OptimizerNames.GALORE_ADAMW_8BIT,
+            OptimizerNames.GALORE_ADAFACTOR,
+        ]:
+            if not is_galore_torch_available():
+                raise ImportError(
+                    "You need to install `galore_torch` in order to use GaLore optimizers"
+                    " install it with `pip install git+https://github.com/jiaweizzhao/GaLore`"
+                )
+
+            from galore_torch import GaLoreAdafactor, GaLoreAdamW, GaLoreAdamW8bit
+
+            optimizer_mapping = {
+                OptimizerNames.GALORE_ADAMW: GaLoreAdamW,
+                OptimizerNames.GALORE_ADAMW_8BIT: GaLoreAdamW8bit,
+                OptimizerNames.GALORE_ADAFACTOR: GaLoreAdafactor,
+            }
+
+            optimizer_cls = optimizer_mapping[args.optim]
+
+            if args.galore_target_modules is None:
+                raise ValueError(
+                    "You need to define a `galore_target_modules` in order to properly use GaLoRe optimizers"
+                )
+
+            if not isinstance(args.galore_target_modules, list):
+                raise ValueError(
+                    f"`galore_target_modules` has to be a list of strings, you passed {args.galore_target_modules}"
+                )
+
+            if model is None:
+                raise ValueError("You need to pass a model in order to correctly initialize a GaLore optimizer.")
+
+            galore_params = []
+            for module_name, module in model.named_modules():
+                if not isinstance(module, nn.Linear):
```

Comment:
I wonder if we should raise an error if the target module name matches but the layer type is not `Linear`. Let's assume that a user matches N linear layers and accidentally 1 other type like `Embedding`, the embedding layer would be ignored but the user doesn't get any error message or warning.

Reply:
This sounds like a good idea!

Reply:
Yes! I think we should pass a warning to be able to pass simple regex such as `.*.attn.*.` - lmk wdyt !